azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/manager/AppServiceKuduManager.java

```diff
@@ -113,16 +130,27 @@ public void deleteFile(String path) {
     }
 
     public List<ProcessInfo> listProcess() {
-        return this.kuduService.listProcess(host).block().getValue();
+        final Mono<Response<List<ProcessInfo>>> responseMono = this.kuduService.listProcess(host);
+        return getValueFromResponseMono(responseMono, Collections.emptyList());
     }
 
     public CommandOutput execute(final String command, final String dir) {
         final CommandRequest commandRequest = CommandRequest.builder().command(command).dir(dir).build();
-        return this.kuduService.execute(host, JsonUtils.toJson(commandRequest)).block().getValue();
+        final Mono<Response<CommandOutput>> responseMono = kuduService.execute(host, JsonUtils.toJson(commandRequest));
+        return getValueFromResponseMono(responseMono);
     }
 
     public TunnelStatus getAppServiceTunnelStatus() {
-        return this.kuduService.getAppServiceTunnelStatus(host).block().getValue();
+        final Mono<Response<TunnelStatus>> responseMono = this.kuduService.getAppServiceTunnelStatus(host);
+        return getValueFromResponseMono(responseMono);
+    }
+
+    private <T> T getValueFromResponseMono(Mono<Response<T>> response) {
+        return getValueFromResponseMono(response, null);
+    }
+
+    private <T> T getValueFromResponseMono(Mono<Response<T>> response, T defaultValue) {
+        return Optional.ofNullable(response.block()).map(Response::getValue).orElse(defaultValue);
```

Comment:
Mono.block will never return null, so two getValueFromResponseMono methods are the same.